src/components/combo_box/combo_box.tsx

```diff
@@ -558,6 +558,10 @@ export class OuiComboBox<T> extends Component<
   };
 
   onComboBoxFocus: FocusEventHandler<HTMLInputElement> = (event) => {
+    if (this.state.hasFocus) {
+      return;
+    }
```

Comment:
This change makes it so that `onFocus` isn't called if combo box is already in focus. Basically, `onFocus` won't be called if you click on combo box again if it's already open.

cc @KrooshalUX for UX guidance on this change

Reply:
Demo'd to @KrooshalUX and got approval for this change.